Added I2C_1 and CAN_1 master for the Nucleo L476RG board.

Josh0027 opened the pull request:
Created Nine Degrees of Freedom Sample Application using MPU6050
	and HMC5883L sensors. Validated on Nucleo L476RG and
	Intel Quark D2000.

Validated the CAN_1 functionality on the Nucleo L476RG.

Signed-off-by: Jose S. Cofreros Jr <josh.cofreros@gmail.com>

mnkp committed:
board: efr32_slwstk6061a: add support for flashing via OpenOCD

The version of OpenOCD provided by Zephyr SDK is too old and does not
contain support for EFR32FG1P chip. This patch assumes that user
compiles and installs the newest version of OpenOCD.

Signed-off-by: Piotr Mienkowski <piotr.mienkowski@gmail.com>

pswarnak committed:
lib: cmsis_rtos_v1: Minor refactor of CMSIS implemetation

Add few missing NULL checks to avoid crash. Also, minor
refactor of signal code and disable osFeature_Wait to
signify osWait function not implemented.

Signed-off-by: Praful Swarnakar <praful.swarnakar@intel.com>

ibisz committed:
scripts: dts: extract: clock.py fix node alias name

A node alias name should be applied to the all property names.

Before this patch, the alias name is applied erroneously
just for CLOCK_CONTROLLER, skipping the CLOCK_BITS and CLOCK_BUS
property names.

Signed-off-by: Istvan Bisz <istvan.bisz@t-online.hu>

cvinayak committed:
usb: device: nrf5: Fix HF clock enable check

Fixed return value check after HF clock is requested in
non-blocking mode.

hci_usb sample on nRF52840 SoC failed to enumerate USB
device on cold reset, the clock_control_on interface would
return -EINPROGRESS when clock is enabled in non-blocking
mode.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

dcpleung committed:
drivers: grove/light: change ADC resolution to 12-bit

The driver itself expects the ADC resolution to be 12-bit.
So set it.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
drivers: grove/temperature: ADC resolution to be 12-bit

The driver itself expects the ADC resolution to be 12-bit.
So set it.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
tests/drivers: adc: add config for Quark D2000

Add the configuration block for Quark D2000 for the adc_api
test.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
samples: grove: re-add Quark D2000 to sample.yaml

Since we now have a working ADC driver for Quark D2000,
add the whitelist back.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
Revert "boards: quark_d2000_crb: Remove adc support"

This reverts commit b0d3b560238fac09e63064620de9ff820cf8cc47.

Since we now have a working ADC driver for Quark D2000,
re-add the tag.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
adc: remove adc_qmsi driver

We now have a native driver for Quark D2000 so the QMSI
one can be removed.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
adc: add native driver for Quark D2000

This adds a native ADC driver for Quark D2000.

Original: origin

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

Josh0027 committed:
Added I2C_1 and CAN_1 master for the Nucleo L476RG board.

Created Nine Degrees of Freedom Sample Application using MPU6050
	and HMC5883L sensors. Validated on Nucleo L476RG and
	Intel Quark D2000.

Validated the CAN_1 functionality on the Nucleo L476RG.

Signed-off-by: Jose S. Cofreros Jr <josh.cofreros@gmail.com>

nashif committed:
xtensa: removed obsolete headers

Went in by mistake in a bad rebase, fixes #9753.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
xtensa: removed obsolete headers

Went in by mistake in a bad rebase, fixes #9753.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
doc: esp32: update flashing instructions

Details about using west for flashing instead of relying on environment
variables.

Fix #9389.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif commented on drivers/sensor/hmc5883l/Kconfig:
```diff
@@ -20,7 +20,7 @@ config HMC5883L_NAME
 	string "I2C master where HMC5883L is connected"
 	depends on HMC5883L
-	default "I2C_0"
+	default "I2C_1"
```
why are you changing this here? If you need that for you sample, you can set to I2C_1 in your sample.

Josh0027 replied:
Reverted to original code and declared in prj.conf.

nashif commented on boards/arm/nucleo_l476rg/pinmux.c:
```diff
@@ -56,6 +56,12 @@ static const struct pin_config pinconf[] = {
 	{STM32_PIN_PC12, STM32L4X_PINMUX_FUNC_PC12_SPI3_MOSI},
 #endif /* CONFIG_SPI_3 */
+
+#ifdef CONFIG_CAN_1
```
unrelated change

Josh0027 replied:
Removed all instances of CAN.

nashif reviewed (changes requested):
please split the changes into multiple commits, can ,sample, dts changes for the boards, etc.

henrikbrixandersen committed:
doc: settings: fix inline sample code

Fix a couple of issues in the code examples for the settings
subsystem.

Signed-off-by: Henrik Brix Andersen <henrik@brixandersen.dk>

Commit:
Bluetooth: Mesh: Fix checking for subnet when recovering App Key

The ordering of items in flash is not guaranteed, so it's possible we
get an App Key before the corresponding Net Key. Remove the check for
a Net Key, since the storing code should never store an App Key if
there is no corresponding Net Key.

Fixes #9670

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

SebastianBoe committed:
cmake: Don't use the environment variable CFLAGS

Some distros set the environment variable CFLAGS, this will
accidentally affect Zephyr builds.

To fix this we clear the environment variable from within the Zephyr
build system for the duration of the CMake execution.

Until now we have been instructing the user to clear it, but it is
easier for the user if we clear it for him.

The same applies to CXXFLAGS.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

cvinayak committed:
drivers: entropy: nrf5: Fix ENTROPY_BUSYWAIT from hanging

Missing clear pending RNG_IRQn when generating multiple
random numbers caused CPU to hang at __WFE call in the
loop waiting for new value.

This fixes commit ddb7f88f9ed5 ("drivers: entropy: nrf5: Fix
ENTROPY_BUSYWAIT implementation")

Fixes #9523.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

AdithyaBaglody committed:
tests: benchmarks: timing_info: Fixed userspace benchmarks for nrf52

In NRF52 we need to write to a capture register to retrieve the current
counter value. This was missing for userspace implementation.

Fixes: GH-9676

Signed-off-by: Adithya Baglody <adithya.nagaraj.baglody@intel.com>

ioannisg committed:
doc: fix minor typo in clock.rst

Remove redundant "is" in the description of Kernel system clock.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

pfalcon committed:
doc: relnotes-1.13: Condense/cleanup networking section

Merged closely related items into one bullet point, improve
description of some items, try to sort items a bit better
(that can be further improved).

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>

rveerama1 committed:
samples: net: Fix out-of-bound access

Payload variable can hold max 64 bytes but value of size can be more
than that. Memsetting using value of size can overrun payload array.
Fixes coverity issue.

Coverity-CID: 187823
Fixes: #9638

Signed-off-by: Ravi kumar Veeramally <ravikumar.veeramally@linux.intel.com>

rveerama1 committed:
net: ipv6: Initialize memory

Fix accessing uninitialized memory. Fixes coverity issue.

Coverity-CID: 187902
Fixes: #9768

Signed-off-by: Ravi kumar Veeramally <ravikumar.veeramally@linux.intel.com>

rveerama1 committed:
samples: net: coap: Remove apps using raw mbedtls APIs

This removes these network sample applications
  samples/net/coaps_client
  samples/net/coaps_server

as they are using low level mbedtls APIs. You should use
preferably socket based or net-app based applications.

Signed-off-by: Ravi kumar Veeramally <ravikumar.veeramally@linux.intel.com>

AdithyaBaglody committed:
subsys: random: xoroshiro128: Use entropy get isr during init

Current implementation of the xoroshiro depends on the ISR being
triggered when the interrupts is locked. This patch proposes
implementing the init with entropy_get_isr. This implementation
can be called at PRE_KERNEL_2 stage, even when the interrupts
are locked.

Fixes: GH-8199

Signed-off-by: Adithya Baglody <adithya.nagaraj.baglody@intel.com>

ulfalizer committed:
scripts: kconfig: Use unique_{defined_syms,choices} and clean up a bit

 - Use Kconfig.unique_{defined_syms,choices} to avoid redundant checks
   for symbols/choices defined in multiple locations. These were
   recently added to Kconfiglib.

 - Remove the comment about the alldefconfig starting state. It probably
   isn't useful here.

 - Print the messages about loading configuration files just before they
   are actually loaded. That looks less confusing if one of them fails
   to load.

 - Line-wrap the error message about non-whitelisted warnings.

 - Misc. other minor code cleanup.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer committed:
scripts: kconfig: Put a blank line before warnings and errors

Since the warnings generated by this script are so spammy, it can be
difficult to tell where they start and end. Put a blank line before each
one to make it clearer.

Also put blank lines around the non-whitelisted-warning error.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

Commit:
arch: xtensa: remove extra #endif

Remove extra #endif that should have been removed when the
corresponding #ifdef CONFIG_KERNEL_EVENT_LOGGER_SLEEP was
removed in commit a2248782a25e205be9b37a03fc47131856ba406d

Signed-off-by: Sathish Kuttan <sathish.k.kuttan@intel.com>

dbkinder committed:
doc: give nicer error if env not set up

If one forgets to source the zephyr-env.sh script before starting a doc
build (using the top-level Makefile), a long screenful of error messages
go by and you can miss what the actual problem is.

Check if ZEPHYR_BASE is not set and give a nice short error message.

(Updated to a more generic error message in anticipation of #9801
deprecating use of zephyr-env.sh)

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

dbkinder committed:
doc: fix reference to log.h

logger api header file reference was incorrect

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

dbkinder committed:
doc: fix reference to props file

Fix reference to misspelled file name in nsim_em documentation

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

carlescufi committed:
Bluetooth: mesh: Fix missing semicolon

Fix a missing semicolon at the end of a statement.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

erwango commented on boards/arm/nucleo_l476rg/Kconfig.defconfig:
```diff
@@ -35,4 +35,7 @@ config PWM_STM32_2
 
 endif # PWM
 
+config I2C_1
```
This line should be under `if SPI` statement.

Josh0027 replied:
Added under "if I2C" instead.

erwango commented on boards/arm/nucleo_l476rg/nucleo_l476rg_defconfig:
```diff
@@ -39,3 +39,7 @@ CONFIG_PWM=y
 CONFIG_STM32_ARM_MPU_ENABLE=y
+
+#enable i2c
+CONFIG_I2C_1=y
```
This is not needed if alredy done in Kconfig.defconfig

Josh0027 replied:
Reverted to previous code.

erwango commented on samples/sensor/nine_degrees_freedom/src/main.c:
```diff
@@ -0,0 +1,92 @@
+ */
+
+/*
+ * 08/23/2018	Jose Sune Cofreros Jr. 	Created nine degrees of freedom 
```
Please reuse SPDX boilerplate

Josh0027 replied:
Done. Followed common boilerplate.

erwango commented on samples/sensor/nine_degrees_freedom/src/main.c:
```diff
@@ -0,0 +1,92 @@
+{
+	struct device *dev1 = device_get_binding("MPU6050");
+	struct device *dev2 = device_get_binding("HMC5883L");
+
```
you should check dev1 and dev2 are not NULL

Josh0027 replied:
Done.

erwango commented on boards/arm/nucleo_l476rg/Kconfig.board:
```diff
@@ -9,3 +9,5 @@
 config BOARD_NUCLEO_L476RG
 	bool "Nucleo L476RG Development Board"
 	depends on SOC_STM32L476XG
+	select HAS_DTS
```
Not needed. Done in arch/arm/code/Kconfig

erwango commented on boards/arm/nucleo_l476rg/Kconfig.board:
```diff
@@ -9,3 +9,5 @@
 	bool "Nucleo L476RG Development Board"
 	depends on SOC_STM32L476XG
+	select HAS_DTS
+	select HAS_DTS_I2C
```
Not needed. Done in `drivers/i2c/Kconfig.stm32`

erwango reviewed (changes requested):
Some additional comments

galak committed:
scripts: dts: extract: Fix handling of irq prop being a list of lists

Before dtc 1.4.7 we'd get something like the following for an interrupt
property in which #interrupt-cells is 2.

interrupts = <1 2 3 4>;

After dtc 1.4.7 we get:

interrupts = <1 2>, <3 4>;

We should handle both cases in the extract interrupt handling code.  So
if we see a list of lists, we flatten it to a single list to normalize
the property.

Fixes: #9558

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

carlescufi committed:
doc: Add Bluetooth release notes for 1.13

Add the relevant highlights from the 1.13 development cycle for the
Bluetooth subsystem and its samples.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

metapsychologe committed:
c++: fix comparison between signed and unsigned types

comparing int i with size_t size generates a sign-compare warning.
This commit solves the issue by giving i the type needed.
As older c standards do not allow to declare i after stack += 4 I
inserted the variable checked_stack that gets the type we need in the
comparison.

Signed-off-by: Alexander Polleti <metapsycholo@gmail.com>

metapsychologe committed:
net: fix return type of net_addr_type2str

As we return stuff that is probably in the flash the return type should
be const char * and not char * as the user better doesn't try to change
them!

Signed-off-by: Alexander Polleti <metapsycholo@gmail.com>

nashif committed:
boards: arduino_101: remove mention of boot stub

The stub is used for testing only and is not compatible anymore with the
usage documented here.

Fixed #9772

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

Commit:
tests: Add kernel/sched/deadline test for EDF validation

Simple test for CONFIG_SCHED_DEADLINE.  It creates a bunch of threads
at randome deadlines but within the same priority, and validates that
they run in the correct order.

Signed-off-by: Andy Ross <andrew.j.ross@intel.com>

nashif committed:
tracing: remove stray event_logger code

Remove obsolete kernel event logger code.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

Commit:
boards: fix up reel_board Kconfig warnings

When I2C and SPI are not used (such as when building a simple
bootloader for this target), the current Kconfig is generating a fair
bit of warning spew. Clean it up by moving some explicit _defconfig
settings into defaults.

Signed-off-by: Marti Bolivar <marti@foundries.io>

fallrisk committed:
adc: sam: Fixed adc_api test for the SAM E70.

The interval_us is supported by the adc_context code. It is not a
feature that the driver writer needs to code to support. Fixes
bug #9723.

Signed-off-by: Justin Watson <jwatson5@gmail.com>

pfalcon committed:
net: net_addr_ntop: Clarify docstring

The string returned is definitely NUL terminated, previously was
apaprently a typo.

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>

pfalcon committed:
CODEOWNERS: Add @pfalcon for drivers/ethernet/

As I'm a reviewer for net/ip/, be sure to review also changes to
drivers/ethernet/, as that's closely related to the IP stack.

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>

jukkar committed:
net: gptp: Set stack size correctly for handler thread

The gptp stack size was not properly set, one must use
the K_THREAD_STACK_SIZEOF() macro to calculate stack size.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

MaureenHelm committed:
doc: release notes: Add arm arch and boards for 1.13

Adds arm highlights to the architectures and boards sections.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

nashif committed:
Revert "kernel: Disable tick handling when !MULTITHREADING"

This reverts commit 8dcd5f8c77c8d5852ab92a3c678b46fd024e388b.

Single thread keep introducing more issues, decided to remove the
feature completely and push any required changes for after 1.13.

See #9808

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
Revert "kernel: Enable interrupts for MULTITHREADING=n on supported arch's"

This reverts commit 17e9d623b417d7a0e9b32ca9b5c4b6cd89062ec0.

Single thread keep introducing more issues, decided to remove the
feature completely and push any required changes for after 1.13.

See #9808

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

MaureenHelm committed:
doc: release notes: Fill in driver section for 1.13

Fills in the drivers section with notable additions and changes.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

nashif committed:
release: wrap lines in release notes

keep all line wrapped.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

Commit:
crypto: tests: Change crypto tests to use Kconfig defined header

While it's possible to define which mbedTLS config header to use, our
samples should use config-tls-generic.h as default because this header
is configurable through Kconfig.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

Commit:
ext: lib: crypto: Update mbedTLS to 2.12.0

Since 2.9.0 release mbedTLS has some minor functional improvements,
security fixes and bug fixed. A list with all changes can be found
in:

./ext/lib/crypto/mbedtls/ChangeLog

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

Commit:
tests: crypto: mbedtls: Fix mbedTLS self test usage

All _self_test functions should be called only if MBEDTLS_SELF_TEST is
enabled.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

Commit:
samples: drivers: crypto: Fix minimum values for mbedTLS

The latest mbedTLS (2.12) release increased resources requirements on
Zephyr, so it was required increasing the minimum SRAM to run this test
and the main stack size.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

dbkinder committed:
doc: add doc changes to 1.13 release notes

Update draft 1.13 release notes with documentation changes (and fix a
spelling typo)

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

dbkinder committed:
doc: fix 404 error page message

The custom 404 error page reports an incorret "url being requested"

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

nashif committed:
doc: update release notes

Update additional section and overview of the release.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
release: bump to 1.13.0-rc3

Update version.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

ioannisg committed:
doc: add TrustZone support in the list of key features

This commit includes the basic support for ARM TrustZone-M
in the list of key features for 1.13 release.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
doc: add detailed description of Armv8-M features in release notes

This commits adds detailed description of Armv8-M features that
are introduced in the 1.13 release.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

Commit:
tests/kernel/sched/deadline: Disable CONFIG_BT

The BT threads are interfering with the main thread priority selection
and the test fails semi-spuriously on NRF5x boards with "threads ran
too soon" (i.e. not an EDF failure per se, but the fact that the new
threads at K_LOWEST_APPLICATION_PRIO are running instead of the test
thread).  This is a reasonable workaround for testing the
SCHED_DEADLINE ordering behavior, though.

Fixes #9843

Signed-off-by: Andy Ross <andrew.j.ross@intel.com>

Josh0027 committed:
nucleo_l476rg: Declared I2C_1 device for commit-id be0ec288cdbc15ffeb942c762d8861dc607cd704.

Signed-off-by: Jose S. Cofreros Jr <josh.cofreros@gmail.com>

Josh0027 committed:
nucleo_l476rg: Declared I2C_1 device for commit-id be0ec288cdbc15ffeb942c762d8861dc607cd704.

Signed-off-by: Jose S. Cofreros Jr <josh.cofreros@gmail.com>

Josh0027 commented:
Hello. Sorry to bother but can anyone reject my last pull request? I suddenly have all of these non-related changes on the code review. I think I made a mistake on not deleting all the other commits while I was following the review request instructions.

Josh0027 commented:
pfalcon, thank you. I will just close this request and create a new one.

jukkar commented:
Instead of closing, you could just have force pushed a rebased branch.

pfalcon commented:
> Instead of closing, you could just have force pushed a rebased branch.

I'm not sure if that will clear folks superfluously and automatically added as reviewers, so that would be a reason to open a new one. But otherwise, yes, force-pushing a new version is always preferred, because it preserves all the discussion already held. @Josh0027, if you decide to open a new one in the end, please be sure to link back to this one. Thanks!

Josh0027 committed:
Merge branch 'nine_degrees_and_can_driver' of https://github.com/Josh0027/zephyr into nine_degrees_and_can_driver

Updated for code review of nine degrees of freedom sample app.

Signed-off-by: Jose S. Cofreros Jr <josh.cofreros@gmail.com>

pfalcon commented:
@Josh0027: So I guess, it's better to close this one after all, because it confuses people who got into the reviewer list. Please submit a new, cleaned one. Thanks.

@dbkinder : Sorry for the hassle, this is the case of "stray commits got into PR by mistake".
